Shared/Store/BaseDataStore.swift

```diff
@@ -186,12 +188,12 @@ class BaseDataStore {
         self.initialized()
     }
 
-    public func get(_ id: String) -> Observable<LoginRecord?> {
-        return self.listSubject
-            .map { items -> LoginRecord? in
-                return items.filter { item in
-                    return item.id == id
-                    }.first
+    public func get(_ id: String) -> LoginRecord? {
+        do {
+            return try self.loginsStorage?.get(id: id)
+        } catch let error {
+            print("Sync15: \(error)")
+            return nil
```

Comment:
AFAICT, this is the crux of the matter. 

It feels quite synchronous _-y_ and MVC _-y_, which makes me nervous. Are there any restrictions you'd like to make on how this method should be called? i.e. if this represents an exception to the rules around how we manage state, how can we make the exception smaller? e.g. perhaps if we expose most of the app to the previous way of getting state, but only use this new synchronous route for _that one time_ when we have too. 

Also: could you comment that this is unusual, and why we're doing it in not the usual way.

Reply:
:-/ this is a great point, and actually doesn't resolve the bug in the first place anyway. I realized this `get` method was there and doing the sql lookup under the hood was almost definitely more performant than what we had before so I was just optimizing. It _should_ be async though, so I'm changing it.